### PR TITLE
Static resource handler switch - WIP

### PR DESCRIPTION
**Pull request by gsteel**
With reference to #49 - It should be easier to disable the serving of static files. Before this change, I'm assuming that you'd need to overwrite the container configuration with a factory designed to return a `null` for the Static Resource Handler, or modify the container configuration at runtime (??)
This 'switch' preserves current behaviour whilst allowing an easier way to disable serving of static files.
If this is a potential solution for #49 I'll update docs and add tests.

- [x] Are you creating a new feature?
  - [x] Why is the new feature needed? What purpose does it serve?
  - [x] How will users use the new feature?
  - [x] Base your feature on the `develop` branch, and submit against that branch.
  - [x] Add only one feature per pull request; split multiple features over multiple pull requests
  - [ ] Add tests for the new feature.
  - [ ] Add documentation for the new feature.
  - [ ] Add a `CHANGELOG.md` entry for the new feature.

### CODE REVIEW

**Comment by wshafer** [comment]
Apparently I am using a much older version of this package, 0.1.1 to be exact, and didn't notice that this had already been abstracted out.

That said, I do like having the enable flag myself.  That's a little more straight forward to me, but the current implementation works too, though I still think for the 3.0 release this should be turned off by default.

File: src/ConfigProvider.php [hunk 1]
```diff
@@ -56,11 +59,12 @@ public function getDependencies() : array
                 PidManager::class                     => PidManagerFactory::class,
                 SwooleRequestHandlerRunner::class     => SwooleRequestHandlerRunnerFactory::class,
                 ServerRequestInterface::class         => ServerRequestSwooleFactory::class,
-                StaticResourceHandlerInterface::class => StaticResourceHandlerFactory::class,
+                StaticResourceHandler::class          => StaticResourceHandlerFactory::class,
```
Comment (weierophinney) [inline src/ConfigProvider.php]: Why this change? What does it accomplish, exactly?

Comment (gsteel) [reply src/ConfigProvider.php]: TBH, Nothing much - it's just  personal habit that crept in - If you want to target that particular class rather than the interface, which might be swapped out for something else, it's more explicit, rather than, fe, attaching a delegator to the interface and having to test if it's an instance of something. That's why, but it's pretty pointless I'll agree!

**Comment by weierophinney** [comment]
@gsteel I've added a test for the new flag, as well as documentation; thanks a ton for doing this!

**Comment by gsteel** [comment]
My pleasure! Thanks for merging.
